lib/widgets/home.dart

```diff
@@ -2,96 +2,528 @@ import 'package:flutter/material.dart';
+    context: context,
+    // Clip.hardEdge looks bad; Clip.antiAliasWithSaveLayer looks pixel-perfect
+    // on my iPhone 13 Pro but is marked as "much slower":
+    //   https://api.flutter.dev/flutter/dart-ui/Clip.html
+    clipBehavior: Clip.antiAlias,
+    useSafeArea: true,
+    isScrollControlled: true,
+    backgroundColor: designVariables.bgBotBar,
+    builder: (BuildContext _) {
+      return PerAccountStoreWidget(
+        accountId: accountId,
+        child: SafeArea(
+          minimum: const EdgeInsets.only(bottom: 8),
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.stretch,
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              Flexible(child: InsetShadowBox(
+                top: 8, bottom: 8,
+                color: designVariables.bgBotBar,
+                child: SingleChildScrollView(
+                  padding: const EdgeInsets.symmetric(vertical: 8, horizontal: 8),
+                  child: Column(children: menuItems)))),
+              const Padding(
+                padding: EdgeInsets.symmetric(horizontal: 16),
+                child: AnimatedScaleOnTap(
+                  scaleEnd: 0.95,
+                  duration: Duration(milliseconds: 100),
+                  child: ActionSheetCancelButton())),
+            ])));
+    });
+}
+
+abstract class _MenuButton extends StatelessWidget {
+  const _MenuButton();
+
+  String label(ZulipLocalizations zulipLocalizations);
+
+  bool get selected => false;
+
+  /// An icon to display before [label].
+  ///
+  /// Must be non-null unless [buildLeading] is overridden.
+  IconData? get icon;
+
+  Widget buildLeading(BuildContext context) {
+    assert(icon != null);
+    final designVariables = DesignVariables.of(context);
+    return Icon(icon, size: 24, color: selected ? designVariables.iconSelected
+                                                : designVariables.icon);
+  }
+
+  void onPressed(BuildContext context);
+
+  void _handlePress(BuildContext context) {
+    // Dismiss the enclosing action sheet immediately,
+    // for swift UI feedback that the user's selection was received.
+    Navigator.of(context).pop();
+
+    onPressed(context);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    final borderSideSelected = BorderSide(width: 1,
+      strokeAlign: BorderSide.strokeAlignOutside,
+      color: designVariables.borderMenuButtonSelected);
+    final buttonStyle = TextButton.styleFrom(
+      padding: const EdgeInsets.symmetric(vertical: 9, horizontal: 8),
+      foregroundColor: designVariables.labelMenuButton,
+      splashFactory: NoSplash.splashFactory,
+      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(10)),
+    ).copyWith(
+      backgroundColor: WidgetStateColor.fromMap({
+        WidgetState.pressed: designVariables.bgMenuButtonActive,
```

Comment:
In dark mode, the pressed button background `bgMenuButtonActive` is black with 20% opacity, and it's on a `#222222` background (`bgBotBar`). So it should be darker than the background. But it's not:

<img width="664" alt="image" src="https://github.com/user-attachments/assets/f24bb043-9a2e-420b-b52b-b12212635e1b">

…well actually it is 🙂 but only for a frame or two before an animation makes it light like this. I think the culprit is a default `overlayColor` value or something like that. I notice that it gets darker if I pass `overlayColor: Colors.transparent`; that may or may not be the solution but it's an observation.

Reply:
I found it helpful to look at the defaults:

```
  /// ## Material 3 defaults
  ///
  /// If [ThemeData.useMaterial3] is set to true the following defaults will
  /// be used:
  ///
  /// {@template flutter.material.text_button.material3_defaults}
  /// * `textStyle` - Theme.textTheme.labelLarge
  /// * `backgroundColor` - transparent
  /// * `foregroundColor`
  ///   * disabled - Theme.colorScheme.onSurface(0.38)
  ///   * others - Theme.colorScheme.primary
  /// * `overlayColor`
  ///   * hovered - Theme.colorScheme.primary(0.08)
  ///   * focused or pressed - Theme.colorScheme.primary(0.1)
  ///   * others - null
  /// * `shadowColor` - Colors.transparent,
```
I guess visually the `pressed` state got affected because `overlayColor` is `Theme.colorScheme.primary(0.1)` by default. We will lose the color distinction for the `hovered`/`focused` state, but we can still control them if we need to.